src/main/java/me/totalfreedom/totalfreedommod/util/History.java

```diff
@@ -0,0 +1,102 @@
+package me.totalfreedom.totalfreedommod.util;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.UUID;
+import org.bukkit.scheduler.BukkitRunnable;
+import org.bukkit.ChatColor;
+import org.bukkit.command.CommandSender;
+import me.totalfreedom.totalfreedommod.TotalFreedomMod;
+import me.totalfreedom.totalfreedomMod.util.FSync;
+
+public class History
+{
+    public static DateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    
+    public static void reportHistory(final CommandSender sender, final String username)
+    {
+        Player player = Bukkit.getPlayer(sender.getName());
```

Comment:
This does not work when the CommandSender is a console. Instead, perhaps change `FSync.playerMsg()` to take a CommandSender as an argument?